src/extension/utils/confFileToInputForm.ts

```diff
@@ -0,0 +1,158 @@
+/* ---------------------------------------------------------------------------------------------
+ *  Convert conf file to input form
+ *-------------------------------------------------------------------------------------------- */
+
+import { nanoid } from 'nanoid'
+import type { InputFormData, ConfFile } from '../types'
+
+const emptyForm: InputFormData = {
+  mainSolidityFile: '',
+  mainContractName: '',
+  specFile: '',
+  solidityCompiler: '',
+  useAdditionalContracts: false,
+  additionalContracts: [],
+  link: [],
+  extendedSettings: [{ id: nanoid(), flag: '' }],
+  useStaging: false,
+  branch: 'master',
+  cacheName: '',
+  message: '',
+  additionalSettings: [
+    {
+      id: nanoid(),
+      option: '',
+      value: '',
+    },
+  ],
+  name: '',
+  solc_map: [],
+}
+
+const stableFields = [
+  'files',
+  'verify',
+  'solc',
+  'link',
+  'settings',
+  'staging',
+  'cache',
+  'msg',
+]
+
+function getAdditionalSettings(confFile: ConfFile) {
+  const copy = { ...confFile }
+
+  stableFields.forEach(field => {
+    delete copy[field]
+  })
+
+  return copy
+}
+
+/**
+ * converts conf file to input form
+ * @param confFile to convert
+ * @param name of the run
+ * @returns input form
+ */
+export function confFileToFormData(
+  confFile: ConfFile,
+  name: string,
+): InputFormData {
+  const form = emptyForm as InputFormData
+
+  form.name = name
+
+  // convert solidity file / contract
+  if (Array.isArray(confFile.files) && confFile.files.length > 0) {
+    form.mainSolidityFile = confFile.files[0] as string
+
+    if (form.mainSolidityFile.includes(':')) {
+      form.mainSolidityFile = form.mainSolidityFile.split(':')[0]
+    }
+
+    if (confFile.files.length > 1) {
+      const [, ...additional] = confFile.files
```

Comment:
why did we add ',' to [, ...additional]

Reply:
the meaning of this is add to the end of the array. in this case it is indeed redundant